### PR TITLE
SPARKC-144: Adds Spark-Packages plugin

### PR DESCRIPTION
**Pull request by RussellSpitzer**
@pkolaczk Since I don't have admin creds on this repo you'll need to do the publish for the 1.2.0 artifacts. It should be as simple as rebasing this branch against the 1.2.0 tag and then running

`sbt spPublish`

With your ~/.ivy2/.credentials containing information like
```
realm=Spark Packages Realm
host=spark-packages.org
user=RussellSpitzer
password=GitHubAuthTokenWithRead:Org
```

I've had to change our license name to match the allowed naming in the Spark-Packages Repo.

### CODE REVIEW

**Comment by pkolaczk** [comment]
What's this:
`Zip File created at: /home/pkolaczk/Projekty/DataStax/spark-cassandra-connector/target/abcdefghi-1.2.0.zip` ?

**Comment by pkolaczk** [comment]
Still not working:
`u"pom file doesn't exist in zip archive or is misnamed. Was looking for spark-cassandra-connector-1.2.1.pom, contents: [u'spark-cassandra-connector-1.2.1-s_2.10.jar', u'spark-cassandra-connector-1.2.1-s_2.10.pom']" `

**Comment by pkolaczk** [comment]
It adds an "s_" in the name that spark-packages site doesn't like.

**Comment by RussellSpitzer** [comment]
> Zip File created at: /home/pkolaczk/Projekty/DataStax/spark-cassandra-connector/target/abcdefghi?-1.2.0.zip ?
No idea, I'll look into it

**Comment by RussellSpitzer** [comment]
> It adds an "s_" in the name that spark-packages site doesn't like.

I noticed it was doing this but I assumed it was part of the convention they were using since I couldn't get far enough to actually publish the artifacts ...

**Comment by RussellSpitzer** [comment]
@brkyvz Help plz :)

**Comment by brkyvz** [comment]
@RussellSpitzer Will ping you once I publish a new release (tonight or tomorrow)!

**Comment by RussellSpitzer** [comment]
Thanks I'll be waiting

**Comment by brkyvz** [comment]
@RussellSpitzer Just published version 0.2.2!

**Comment by RussellSpitzer** [comment]
@brkyvz I appreciate it! I'll see if i can get a new build going tonight

File: project/Settings.scala [hunk 1]
```diff
@@ -44,6 +46,15 @@ object Settings extends Build {
     versionStatus        := Versions.status(scalaVersion.value, scalaBinaryVersion.value)
   )
 
+  lazy val sparkPackageSettings = Seq(
```
Comment (brkyvz) [inline project/Settings.scala]: I would suggest adding (`++`) this to `settings` on L58 instead of `assembledSettings`

**Comment by brkyvz** [comment]
then you can try publishing with `sbt/sbt spark-cassandra-connector/spDist`
Since this is a multi-project build, running `sbt/sbt spDist` will try to upload every project (even the demos).

File: project/Settings.scala (hunk 1 above)
Comment (helena) [inline project/Settings.scala]: @brkyvz Doing that would mean you want everything to be under assembly, and we do not.

**Comment by helena** [comment]
@RussellSpitzer I have not tested it in the branch but the config looks good

File: project/plugins.sbt [hunk 2]
```diff
@@ -17,3 +15,9 @@ resolvers += "sonatype-releases" at "https://oss.sonatype.org/content/repositori
 resolvers += Classpaths.sbtPluginReleases
 
 addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.0.4")
+
+// Spark Packages Plugin
+resolvers += "Spark Package Main Repo" at "https://dl.bintray.com/spark-packages/maven"
```
Comment (brkyvz) [inline project/plugins.sbt]: You won't need this with 0.2.2

**Comment by RussellSpitzer** [comment]
@pkolaczk Want to try to publish our Packages again?

**Comment by RussellSpitzer** [comment]
retest this please

**Comment by RussellSpitzer** [comment]
Nvm something is broken...

**Comment by RussellSpitzer** [comment]
@brkyvz without the old bintray resolver i can't grab the spark-packages plugin you may want to add it back to the README, or I may be missing something :)

**Comment by pkolaczk** [comment]
I noticed it slightly breaks sbt-assembly:
```
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
[info] ScalaTest
[info] Run completed in 2 seconds, 5 milliseconds.
[info] Total number of tests run: 0
[info] Suites: completed 0, aborted 0
[info] Tests: succeeded 0, failed 0, canceled 0, ignored 0, pending 0
[info] No tests were executed.
[info] Passed: Total 44, Failed 0, Errors 0, Passed 44
[info] Checking every *.class/*.jar file's SHA-1.
[info] Merging files...
[warn] Merging 'META-INF/MANIFEST.MF' with strategy 'discard'
[warn] Strategy 'discard' was applied to a file
[info] SHA-1: 9daa9ddd7dcadf8f392d94e1dad530460dac168e
[info] Packaging /home/pkolaczk/Projekty/DataStax/spark-cassandra-connector/spark-cassandra-connector-java/target/scala-2.10/spark-cassandra-connector-java-assembly-1.2.5-SNAPSHOT.jar ...
[info] Done packaging.
[info] Checking every *.class/*.jar file's SHA-1.
[info] Merging files...
[warn] Merging 'NOTICE' with strategy 'rename'
[warn] Merging 'about_files/LICENSE.txt' with strategy 'rename'
[warn] Merging 'about_files/NOTICE.txt' with strategy 'rename'
[warn] Merging 'META-INF/NOTICE.txt' with strategy 'rename'
[warn] Merging 'META-INF/NOTICE' with strategy 'rename'
[warn] Merging 'org/xerial/snappy/native/README' with strategy 'rename'
[warn] Merging 'META-INF/license' with strategy 'rename'
[warn] Merging 'about.html' with strategy 'rename'
[warn] Merging 'META-INF/LICENSE.txt' with strategy 'rename'
[warn] Merging 'META-INF/README.txt' with strategy 'rename'
[warn] Merging 'LICENSE.txt' with strategy 'rename'
[warn] Merging 'META-INF/LICENSE' with strategy 'rename'
[warn] Merging 'LICENSE' with strategy 'rename'
[warn] Merging 'META-INF/DEPENDENCIES' with strategy 'discard'
[warn] Merging 'META-INF/ECLIPSEF.RSA' with strategy 'discard'
[warn] Merging 'META-INF/ECLIPSEF.SF' with strategy 'discard'
[warn] Merging 'META-INF/INDEX.LIST' with strategy 'discard'
[warn] Merging 'META-INF/MANIFEST.MF' with strategy 'discard'
java.lang.RuntimeException: Please add any Spark dependencies by supplying the sparkVersion and sparkComponents. Please remove: org.apache.spark:spark-streaming:1.2.2
	at scala.sys.package$.error(package.scala:27)
	at sbtsparkpackage.SparkPackagePlugin$$anonfun$12$$anonfun$apply$1.apply(SparkPackagePlugin.scala:108)
	at sbtsparkpackage.SparkPackagePlugin$$anonfun$12$$anonfun$apply$1.apply(SparkPackagePlugin.scala:96)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.immutable.List.foreach(List.scala:318)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractTraversable.map(Traversable.scala:105)
	at sbtsparkpackage.SparkPackagePlugin$$anonfun$12.apply(SparkPackagePlugin.scala:96)
	at sbtsparkpackage.SparkPackagePlugin$$anonfun$12.apply(SparkPackagePlugin.scala:94)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:235)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

sbt-assembly does generate the assembly properly, but then fails when trying to process the other modules. Can we just disable the spark-packages plugin for everything other than spark-cassandra-connector and spark-cassandra-connector-java?

**Comment by RussellSpitzer** [comment]
@pkolaczk I figured that out before, I was able to specifically exclude the other projects i'll see if I can figure that out later

**Comment by pkolaczk** [comment]
addressed by #845

**Pull request closed without merging**